director/origin_api_test.go

```diff
@@ -188,3 +193,182 @@ func TestNamespaceKeysCacheEviction(t *testing.T) {
+	require.NoError(t, err, "Failed to convert new key to JWK")
+	err = jwk.AssignKeyID(newKey)
+	require.NoError(t, err)
+	err = newKey.Set(jwk.AlgorithmKey, jwa.ES256)
+	require.NoError(t, err)
+
+	newPublicKey, err := newKey.PublicKey()
+	require.NoError(t, err, "Failed to create new public key")
+
+	// Track how many times the JWKS endpoint is called
+	jwksCallCount := 0
+	var currentKey jwk.Key = oldPublicKey
+	var registryServerURL string
+
+	// Mock registry server that serves JWKS
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		if req.Method == "POST" && req.URL.Path == "/api/v1.0/registry/checkNamespaceStatus" {
+			res := server_structs.CheckNamespaceStatusRes{Approved: true}
+			resByte, err := json.Marshal(res)
+			require.NoError(t, err)
+			w.WriteHeader(http.StatusOK)
+			_, _ = w.Write(resByte)
+		} else if req.URL.Path == "/api/v1.0/registry/test-namespace/.well-known/openid-configuration" {
+			// Return openid-configuration pointing to JWKS
+			jwksUrl := fmt.Sprintf("%s/api/v1.0/registry/test-namespace/.well-known/issuer.jwks", registryServerURL)
+			config := map[string]string{
+				"jwks_uri": jwksUrl,
+			}
+			configByte, err := json.Marshal(config)
+			require.NoError(t, err)
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusOK)
+			_, _ = w.Write(configByte)
+		} else if req.URL.Path == "/api/v1.0/registry/test-namespace/.well-known/issuer.jwks" {
+			jwksCallCount++
+			// Return the current key (starts with old key, can be updated to new key)
+			jwks := jwk.NewSet()
+			err := jwks.AddKey(currentKey)
+			require.NoError(t, err)
+			jwksByte, err := json.Marshal(jwks)
+			require.NoError(t, err)
+			w.Header().Set("Content-Type", "application/json")
+			w.WriteHeader(http.StatusOK)
+			_, _ = w.Write(jwksByte)
+		} else {
+			w.WriteHeader(http.StatusNotFound)
+		}
+	}))
+	defer ts.Close()
+	registryServerURL = ts.URL
+
+	// Spin up mock federation discovery endpoint with embedded mock registry URL.
+	fedInfo := pelican_url.FederationDiscovery{RegistryEndpoint: registryServerURL}
+	test_utils.MockFederationRoot(t, &fedInfo, nil)
+
+	// Initialize director
+	tDir := t.TempDir()
+	kDir := filepath.Join(tDir, "t-issuer-keys")
+	viper.Set(param.IssuerKeysDirectory.GetName(), kDir)
+	viper.Set("ConfigDir", tDir)
+	err = config.InitServer(ctx, server_structs.DirectorType)
+	require.NoError(t, err)
+
+	// Start the TTL cache
+	LaunchTTLCache(ctx, egrp)
+
+	// Get the namespace issuer URL and JWKS URL
+	issuerUrl, err := server_utils.GetNSIssuerURL("/test-namespace")
+	require.NoError(t, err)
+	keyLoc, err := server_utils.GetJWKSURLFromIssuerURL(issuerUrl)
+	require.NoError(t, err)
+
+	// Create a token signed with the old key
+	advTokenCfg := token.NewWLCGToken()
+	advTokenCfg.Lifetime = time.Minute
+	advTokenCfg.Issuer = issuerUrl
+	advTokenCfg.Subject = "test-cache"
+	advTokenCfg.AddAudiences("https://director.example.com")
```

Comment:
It's a mistake that we don't enforce the audience when the Director verifies the advertise token -- let's not fix that here, but we should open another issue for it. In the mean time, this test should use an audience that corresponds to the Director configured via the mock fed root because once we fix that, it'll cause this test to fail otherwise.